pandas/io/common.py

```diff
@@ -1136,18 +1130,20 @@ def _is_binary_mode(handle: FilePath | BaseBuffer, mode: str) -> bool:
     )
 
 
-def _get_binary_io_classes() -> tuple[type, ...]:
+def _get_binary_io_classes(_cache: list[type] = []) -> tuple[type, ...]:
```

Comment:
How about?
```py
@functools.cache
def _get_binary_io_classes() -> tuple[type, ...]:
```

Reply:
Python 3.9+ only

Reply:
`@functools.lru_cache` should work as well, or?

Reply:
Sure if you prefer that over the custom memoization I'm happy to change it

Reply:
There is only one "least recently used" input argument (might be a bit too much to use lru_cache) but it keeps the code slightly simpler. Will defer to @jreback